lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1069,15 +1078,69 @@ SPIRVValue *SPIRVModuleImpl::addNullConstant(SPIRVType *Ty) {
 
 SPIRVValue *SPIRVModuleImpl::addCompositeConstant(
     SPIRVType *Ty, const std::vector<SPIRVValue *> &Elements) {
+  SPIRVConstantComposite *Res;
+  const size_t MaxNumElements = 65535 - 3;
+  const size_t NumElements = Elements.size();
+  if (NumElements > MaxNumElements &&
+      isAllowedToUseExtension(ExtensionID::SPV_INTEL_long_constant_composite)) {
+    for (uint64_t J = 0; J < NumElements; J += MaxNumElements) {
+      auto Start = Elements.begin() + J;
+      auto End = ((J + MaxNumElements) > NumElements) ? Elements.end()
+                                                      : Start + MaxNumElements;
+      std::vector<SPIRVValue *> Slice(Start, End);
+      if (J == 0) {
+        Res = static_cast<SPIRVConstantComposite *>(
+            addCompositeConstant(Ty, Slice));
+      } else {
+        SPIRVConstantComposite::ContinuedInstType Continued =
+            static_cast<SPIRVConstantComposite::ContinuedInstType>(
+                addCompositeConstantContinuedINTEL(Slice));
+        Res->addContinuedInstruction(Continued);
+      }
+    }
+    return Res;
+  }
   return addConstant(new SPIRVConstantComposite(this, Ty, getId(), Elements));
 }
 
+SPIRVEntry *SPIRVModuleImpl::addCompositeConstantContinuedINTEL(
+    const std::vector<SPIRVValue *> &Elements) {
+  return add(new SPIRVConstantCompositeContinuedINTEL(this, Elements));
+}
+
 SPIRVValue *SPIRVModuleImpl::addSpecConstantComposite(
     SPIRVType *Ty, const std::vector<SPIRVValue *> &Elements) {
+  SPIRVSpecConstantComposite *Res;
```

Comment:
I'd prefer to move this variable closer to its usages. Or even get rid of it at all.

Reply:
Moved it closer.